tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/XUnit/XUnitTestOutputHelperQueueTestOutputIntegration.cs

```diff
@@ -58,52 +60,97 @@ internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, string
+                }
+
+                writer.WritePropertyName("timestamp", escape: false);
+                writer.WriteValue(DateTimeOffset.UtcNow.ToUnixTimeNanoseconds() / 1_000_000);
+
+                writer.WritePropertyName("message", escape: false);
+                writer.WriteValue(_message);
+
+                var env = formatter.Env;
+                var service = formatter.Service;
+                if (_span is { } span)
+                {
+                    if (span.GetTag(Tags.Env) is { } spanEnv)
                     {
-                        if (state.HasValue)
-                        {
-                            writer.WritePropertyName("dd.trace_id");
-                            writer.WriteValue($"{state.Value.TraceId}");
-                            writer.WritePropertyName("dd.span_id");
-                            writer.WriteValue($"{state.Value.SpanId}");
-                            writer.WritePropertyName("_dd.origin");
-                            writer.WriteValue($"{state.Value.Origin}");
-                        }
-
-                        return default;
-                    });
-            }
+                        env = spanEnv;
+                    }
 
-            private readonly struct Context
-            {
-                public readonly ulong TraceId;
-                public readonly ulong SpanId;
-                public readonly string Origin;
+                    if (!string.IsNullOrEmpty(span.ServiceName))
+                    {
+                        service = span.ServiceName;
+                    }
 
-                public Context(ulong traceId, ulong spanId, string origin)
+                    writer.WritePropertyName("dd.trace_id", escape: false);
+                    writer.WriteValue($"{span.TraceId}");
+
+                    writer.WritePropertyName("dd.span_id", escape: false);
+                    writer.WriteValue($"{span.SpanId}");
+
+                    if (span.GetTag(TestTags.Suite) is { } suite)
+                    {
+                        writer.WritePropertyName(TestTags.Suite, escape: false);
+                        writer.WriteValue(suite);
+                    }
+
+                    if (span.GetTag(TestTags.Name) is { } name)
+                    {
+                        writer.WritePropertyName(TestTags.Name, escape: false);
+                        writer.WriteValue(name);
+                    }
+
+                    if (span.GetTag(TestTags.Bundle) is { } bundle)
+                    {
+                        writer.WritePropertyName(TestTags.Bundle, escape: false);
+                        writer.WriteValue(bundle);
+                    }
+                }
+
+                writer.WritePropertyName("service", escape: false);
+                writer.WriteValue(service);
+
+                // spaces are not allowed inside ddtags
+                env = env.Replace(" ", string.Empty);
+                env = env.Replace(":", string.Empty);
```

Comment:
Can the env value be cleaned up earlier so we don't have to clean it up here?

Reply:
I added a cache layer in the refactor at https://github.com/DataDog/dd-trace-dotnet/commit/17e24d81ff05bc7a5b9e6987ec43680b365bdf06